apps/meteor/app/api/server/v1/users.ts

```diff
@@ -815,11 +817,32 @@ API.v1.addRoute(
 	{ authRequired: true, validateParams: isUsersAutocompleteProps },
 	{
 		async get() {
-			const { selector } = this.queryParams;
+			const { selector: selectorRaw } = this.queryParams;
+
+			let selector: { exceptions: Required<IUser>['username'][]; conditions: Filter<IUser>; term: string } = {
+				exceptions: [],
+				conditions: {},
+				term: '',
+			};
+
+			try {
+				selector = JSON.parse(selectorRaw);
+
+				if (selector) {
+					selector = clean(selector) as typeof selector;
```

Comment:
from what I can see, `clean` will remove every key starting with `$`, but on `isValidQuery` called below you're allowing `$or` and `$and`, which won't ever happen.

so you either want:
- not call `clean` here (my suggestion)
- remove `isValidQuery` call since there will be no special selector

Reply:
There is a bug in `clean` so this is actually not happening. I'm adding extra tests to ensure the outcome is the expected one.